bootloader/grub.go

```diff
@@ -431,33 +433,28 @@ func staticCommandLineForGrubAssetEdition(asset string, edition uint) string {
 	return string(cmdline)
 }
 
-// modeArgsForGrub validates the snapd mode arguments for the kernel command
-// line and splits them into individual arguments.
-func modeArgsForGrub(modeArgs string, recoveryAsset bool) ([]string, error) {
-	args, err := strutil.KernelCommandLineSplit(modeArgs)
-	if err != nil {
-		return nil, err
-	}
-	// see grub.cfg and grub-recovery.cfg assets, the order is:
-	//      for run mode: snapd_recovery_mode=run <args>
-	// for recovery mode: snapd_recovery_mode=recover snapd_recovery_system=<label> <args>
+// validateModeArgsForGrub validates the snapd mode arguments for the kernel
+// command line.
+func validateModeArgsForGrub(modeArg, systemArg string, recoveryAsset bool) error {
```

Comment:
do we need this? boot passes a lot of things to bootloaders and we don't have validation code for them

Reply:
This is dropped now.